client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -96,12 +98,12 @@ public synchronized int register(String subject, Schema schema)
   }
 
   @Override
-  public synchronized Schema getByID(int id) throws IOException, RestClientException {
-    return getBySubjectAndID(null, id);
+  public synchronized Schema getById(int id) throws IOException, RestClientException {
```

Comment:
I don't think we've ever been explicit about it being public API, but people treat these as such. So this might break other people's code. Possibly add these APIs and deprecate the old ones while leaving the exception so we can remove them whenever the next major version bump happens?